### PR TITLE
Always initialize oneAPI in CI workflows

### PR DESCRIPTION
**Pull request by pbchekin**
It's simpler to initialize oneAPI for each step in CI workflows by default than do it for each individual step that requires it. Also it brings the CI environment in sync with development environments where oneAPI, usually, is initialized in the very begining,

### CODE REVIEW

**Review by gshimansky — APPROVED** [review]
I cannot say that I like this solution very much. It looks more like a hack to be.  It is very easy to miss this setting at the beginning of the very long scenario and therefore have no clue that oneapi is activated. I would like it more to see CI scenarios reusing some shell script that could be run standalone to create a development environment.

**Comment by pbchekin** [comment]
> I would like it more to see CI scenarios reusing some shell script that could be run standalone to create a development environment.

Yes. Currently we are trying to keep our shell scripts in sync with respect to the logic and flow, and it will require some work on the scripts to make it possible to use them in ci workflows, but i think we are moving to the right direction.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -102,6 +102,9 @@ jobs:
       max-parallel: 2
       fail-fast: false
     timeout-minutes: 720
+    defaults:
+      run:
+        shell: bash -noprofile --norc -eo pipefail -c "source /home/runner/intel/oneapi/setvars.sh > /dev/null; source {0}"
```
Comment (leshikus) [inline .github/workflows/e2e.yml:107]: I like adding pipefail on the top level

**Review by leshikus — COMMENTED** [review]
LGTM

**Pull request merged**